### PR TITLE
chore: use Chrome Canary for BiDi testing

### PR DESCRIPTION
**Pull request by Lightning00Blade**
@OrKoN I added the code from Chromium-BiDi for creating a update PR, comment if you want to keep that part.
Keeping it Draft till then as it will need the proper Secret Key set-up

### CODE REVIEW

File: tools/install-browser.mjs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ *  - `node install-browser.mjs /tmp/cache`
+ */
+
+import fs from 'fs/promises';
+
+import actions from '@actions/core';
+import {install, computeExecutablePath} from '@puppeteer/browsers';
+
+try {
+  const browserSpec = (await fs.readFile('.browser', 'utf-8')).trim();
```
Comment (OrKoN) [inline tools/install-browser.mjs]: let's modify it to be `test/.bidi-chrome-browser`?

File: .browser [hunk 2]
```diff
@@ -0,0 +1 @@
+chrome@115.0.5785.0
```
Comment (OrKoN) [inline .browser]: remove?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -151,23 +151,42 @@ jobs:
...
+        if: ${{ matrix.suite == 'chrome-bidi' }}
+        id: browser
+        run: |
+          bidi_chrome_version=$(cat ./node_modules/chromium-bidi/.browser) &&
+          executable_path=$(npx @puppeteer/browsers install $bidi_chrome_version --path ~/.cache/puppeteer/chrome-canary) &&
+          echo "executablePath=$executable_path" >> $GITHUB_OUTPUT
       - name: Tests types
         run: npm run test-types
-      - name: Run all tests (for non-Linux)
+      - name: Run all tests (for non-Linux)bidi_chrome_version
```
Comment (OrKoN) [inline .github/workflows/ci.yml]: accidental change?

Comment (Lightning00Blade) [reply .github/workflows/ci.yml]: Yes indeed.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -151,23 +151,42 @@ jobs:
...
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/puppeteer/chrome-canary
+          key: ${{ runner.os }}-Chrome-Canary-${{ hashFiles('package.json') }}
+      - name: Install Chrome Canary
+        if: ${{ matrix.suite == 'chrome-bidi' }}
+        id: browser
+        run: |
+          bidi_chrome_version=$(cat ./node_modules/chromium-bidi/.browser) &&
+          executable_path=$(cut -d' ' -f2 <<< $(npx @puppeteer/browsers install $bidi_chrome_version --path ~/.cache/puppeteer/chrome-canary)) &&
```
Comment (OrKoN) [inline .github/workflows/ci.yml]: I do not think -f2 is correct, probably it as to be `-f2-` as the executable path might have spaces

Comment (Lightning00Blade) [reply .github/workflows/ci.yml:173]: I think `-f2` is correct as my
```bash
echo $(cut -d'.' -f2- <<<  '.s.2..') --> s.2..
echo $(cut -d'.' -f2 <<<  '.s.2..') --> s
```

**Review by OrKoN — APPROVED** [review]

**Review by wazirarif — COMMENTED** [review]
[changes]( #10261 )

**Pull request merged**